tests/unit/src/commands/initialize/initDeployment.ts

```diff
@@ -0,0 +1,211 @@
+		setEnvironmentVariablesStub = stub();
+		createKeyStub = stub();
+		deleteKeyStub = stub();
+
+		Travis = class {
+			constructor() {}
+			isAuthorized: SinonStub = isAuthorizedStub;
+			createAuthorization: SinonStub = travisCreateAuthorizationStub;
+			deleteAuthorization: SinonStub = travisDeleteAuthorizationStub;
+			fetchRepository: SinonStub = fetchRepositoryStub;
+		};
+
+		GitHub = class {
+			constructor() {}
+			toString() {
+				return 'repo';
+			}
+			createKey: SinonStub = createKeyStub;
+			deleteKey: SinonStub = deleteKeyStub;
+		};
+
+		TravisSpy = spy(Travis);
+	},
+
+	after() {
+		cleanupModuleMocks();
+	},
+
+	beforeEach() {
+		isAuthorizedStub.returns(true);
+		keyFileStub.returns('keyFile');
+		encryptedKeyFileStub.returns('encryptedKeyFile');
+		existsSyncStub.returns(true);
+		fetchRepositoryStub.returns(Promise.resolve({
+			listEnvironmentVariables: listEnvironmentVariablesStub,
+			setEnvironmentVariables: setEnvironmentVariablesStub
+		}));
+		listEnvironmentVariablesStub.returns(Promise.resolve([
+			{ name: 'decryptKey', value: 'decryptKey', isPublic: false },
+			{ name: 'decryptIv', value: 'decryptIv', isPublic: false }
+		]));
+		createDeployKeyStub.returns({
+			publicKey: 'publicKey',
+			privateKey: 'privateKey',
+			encryptedKey: {
+				key: 'encryptedKey',
+				iv: 'encryptedIv'
+			}
+		});
+
+		initDeployment = loadModule('src/commands/initialize/initDeployment', {
+			'../../util/Travis': { default: TravisSpy },
+			'../../util/environment': {
+				decryptKeyName: 'decryptKey',
+				decryptIvName: 'decryptIv',
+				githubAuth: githubAuthStub,
+				keyFile: keyFileStub,
+				encryptedKeyFile: encryptedKeyFileStub
+			},
+			'./createDeployKey': { default: createDeployKeyStub },
+			'@dojo/shim/array': {
+				find: findStub
+			},
+			'fs': {
+				existsSync: existsSyncStub,
+				readFileSync: readFileSyncStub
+			}
+		});
+	},
+
+	afterEach() {
+		TravisSpy.reset();
+		githubAuthStub.reset();
+		keyFileStub.reset();
+		encryptedKeyFileStub.reset();
+		createDeployKeyStub.reset();
+		findStub.reset();
+		existsSyncStub.reset();
+		readFileSyncStub.reset();
+		isAuthorizedStub.reset();
+		travisCreateAuthorizationStub.reset();
+		travisDeleteAuthorizationStub.reset();
+		fetchRepositoryStub.reset();
+		listEnvironmentVariablesStub.reset();
+		setEnvironmentVariablesStub.reset();
+		createKeyStub.reset();
+		deleteKeyStub.reset();
+	},
+
+	'initDeployment': (() => {
+		return {
+			async 'explicit Travis instance and options'() {
+				const travis = new Travis();
+
+				await assertInitDeployment(travis, {
+					deployKeyFile: 'deploy-key.file',
+					encryptedKeyFile: 'deploy-key.enc'
+				});
+			},
+
+			async 'default instance and options; Travis is not authorized, should not create deploy key'() {
+				isAuthorizedStub.returns(false);
+				travisCreateAuthorizationStub.returns(Promise.resolve());
+
+				await assertInitDeployment();
+
+				assert.isTrue(TravisSpy.calledOnce);
```

Comment:
Can we add comments (or create functions) to blocks like this that describe what is being checked. I feel like since there's 4 assertions together that it's testing something, but I'd have to look in code to verify